express-mock-af/services/reports.service.js

```diff
@@ -1,162 +1,208 @@
+
+    return chain(reports)
+      .map((report) => {
+        return report;
+      })
+      .value();
+  }
+
+  /**
+   * Returns an overview of all the metrics for the given provisionSessionIds
+   *
+   * @param reports
+   * @param queryFilter
+   * @returns {Promise<(any)[]>}
+   */
+  async overviewMetricsReport(reports, queryFilter) {
+    const { orderProperty, offset, limit, sortingOrder } = defaults(
+      queryFilter,
+      {
+        orderProperty: "reportTime",
+        sortingOrder: "desc",
+      }
+    );
+
+    return chain(reports)
+      .map((report) => {
+        const receptionReport = pick(report.ReceptionReport, [
+          "clientID",
+          "contentURI",
+        ]);
+
+        const qoeReport = pick(report.ReceptionReport.QoeReport, [
+          "reportPeriod",
+          "reportTime",
+          "recordingSessionId",
+        ]);
+
+        const qoeMetric = report.ReceptionReport.QoeReport.QoeMetric;
+        const availableMetrics = Array.isArray(qoeMetric)
+          ? qoeMetric.map((metric) => {
+              return Object.keys(metric)[0];
             })
-        )).filter(content => content.length !== 0).flat();
-        const transformedJsonResponse = this.transformJSONtoReport(readContent);
-
-        return await this.overviewConsumptionReport(transformedJsonResponse, queryFilter);
-    }
-
-    async overviewConsumptionReport(reports, queryFilter) {
-        const { } = defaults(
+          : [];
+
+        return defaults({}, receptionReport, qoeReport, { availableMetrics });
+      })
+      .orderBy(orderProperty, sortingOrder)
+      .thru((chainInstance) => {
+        let result = chain(chainInstance);
+        if (offset !== undefined) {
+          result = result.drop(offset);
+        }
+        if (limit !== undefined) {
+          result = result.take(limit);
+        }
+        return result;
+      })
+      .value();
+  }
+
+  /**
+   * Filters reports based on the query parameters
+   *
+   * @param reportsList
+   * @param queryFilter
+   * @returns {*}
+   */
+  filterReports(reportsList, queryFilter) {
+    const clearQueryFilter = omitBy(queryFilter, isNil);
+    return reportsList.filter((report) => {
+      if (report.ReceptionReport) {
+        const flattenedReport = merge(
+          report.ReceptionReport,
+          report.ReceptionReport.QoeReport
         );
-
-        return chain(reports)
-            .map((report) => {
-                return report;
-            }).value();
-    }
-
-    /**
-     * Returns an overview of all the metrics for the given provisionSessionIds
-     *
-     * @param reports
-     * @param queryFilter
-     * @returns {Promise<(any)[]>}
-     */
-    async overviewMetricsReport(reports, queryFilter) {
-        const { orderProperty, offset, limit, sortingOrder } = defaults(
-            queryFilter,
-            {
-                orderProperty: 'reportTime',
-                sortingOrder: 'desc'
-            }
+        return (
```

Comment:
In dieser Datei hab ich nur dieses Return Statement geändert. Alles andere habe ich nicht berührt. Sieht nur wegen merge komisch aus

Reply:
Das müsste @KonsumGandalf  reviewen